File: lesson-12/js/program.js
class Node {
    constructor(value) {
        this.value = value;
        this.children = [];

    }

    append = (node) => {
        this.children.push(node);
    }
}

const print = (root, sp = "") => {
    if (root !== null) {
        console.log(sp + root.value);

        if (root.children.length !== 0) {
            root.children.forEach(tree => {
                print(tree, sp + "  ");
            });
        }
    }
}

const city2 = new Node(2);
const city6 = new Node(6);
const city5 = new Node(5);
const city4 = new Node(4);
const city3 = new Node(3);

const city1 = new Node(1);

city2.append(city1);
city2.append(city5);

city5.append(city3);
city5.append(city6);
city5.append(city4);


print(city2);

File: lesson-12/java/Program.java

import java.util.ArrayList;

public class Program {

    public static void main(String[] args) {

        Node city2 = new Node(2);
        Node city6 = new Node(6);
        Node city5 = new Node(5);
        Node city4 = new Node(4);
        Node city3 = new Node(3);

        Node city1 = new Node(1);

        city2.append(city1);
        city2.append(city5);

        city5.append(city3);
        city5.append(city6);
        city5.append(city4);

        print(city2, "");
    }

    public static void print(Node root, String sp) {
        if (root != null) {
            System.out.println(sp + root.value);

            if (root.children.size() != 0) {
                for (int i = 0; i < root.children.size(); i++) {
                    print(root.children.get(i), sp + "  ");
                }
            }
        }
    }
}

class Node {
    int value;
    public ArrayList<Node> children;

    public Node(int value) {
        this.value = value;
        this.children = new ArrayList<>();
    }

    public void append(Node node) {
        this.children.add(node);
    }

}
